### PR TITLE
Branch tab module :(

### LINKED ISSUE #74: MODULES tab - Coding - Adding and deleting modules

### LINKED ISSUE #75: MODULES tab - UI - Show all tasks associated to each module code

### PR DESCRIPTION
**Pull request by ethan-l-m-e**

### CODE REVIEW

File: src/main/java/seedu/tr4cker/logic/commands/ModuleCommand.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+     */
+    public ModuleCommand() {
+        this.toAdd = null;
+        this.toDeleteIndex = null;
+    }
+
+    /**
+     * Constructor for ModuleCommand when user wants to add a module.
+     */
+    public ModuleCommand (Module toAdd) {
```
Comment (ruilingk) [inline src/main/java/seedu/tr4cker/logic/commands/ModuleCommand.java]: I think there should not be a space after `ModuleCommand`?
eg. `public ModuleCommand(Module toAdd)`?

Same goes for the constructor for deletion below.

File: src/main/java/seedu/tr4cker/model/ModelManager.java [hunk 2]
```diff
@@ -45,6 +47,7 @@ public ModelManager(ReadOnlyTr4cker tr4cker, ReadOnlyUserPrefs userPrefs) {
             x -> !Deadline.isFutureDeadline(x.getDeadline().toString()) && !x.isCompleted()));
         filteredCompletedTasks = new FilteredList<>(this.tr4cker.getTaskList().filtered(
             Task::isCompleted));
+        filteredModules = new FilteredList<>(this.tr4cker.getModuleList().filtered(x->true));
```
Comment (ruilingk) [inline src/main/java/seedu/tr4cker/model/ModelManager.java]: Maybe can consider adding a space for `x -> true` for better readability?

File: src/main/java/seedu/tr4cker/model/module/UniqueModuleList.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+     * The module must exist in the list.
+     */
+    public void remove(Module toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new ModuleNotFoundException();
+        }
+    }
+
+    public void setModules(UniqueModuleList replacement) {
```
Comment (ruilingk) [inline src/main/java/seedu/tr4cker/model/module/UniqueModuleList.java:74]: Missing JavaDocs for public method.

**Review by ruilingk — COMMENTED** [review]
Some minor changes and it LGTM.

File: src/main/java/seedu/tr4cker/model/module/UniqueModuleList.java (hunk 3 above)
Comment (ethan-l-m-e) [reply src/main/java/seedu/tr4cker/model/module/UniqueModuleList.java:74]: image

Idk why it doesn't show its been updated

Comment (ruilingk) [reply src/main/java/seedu/tr4cker/model/module/UniqueModuleList.java:74]: oh can already!

**Review by ruilingk — APPROVED** [review]
lgtm!

**Pull request merged**